include/gz/sensors/Sensor.hh

```diff
@@ -231,6 +232,30 @@ namespace gz
       /// \return True if there are subscribers, false otherwise
       public: virtual bool HasConnections() const;
 
+      /// \brief Set the sensor in triggered mode. In this mode,
+      /// - the sensor will only update if a new message has been published to
+      ///    the passed _trigger_topic since the last update,
+      /// - until the next message is published on _trigger_topic, all Update
+      ///   calls will return false, and
+      /// - update_rate is forced to zero.
+      /// \param[in] _trigger_topic The topic on which the sensor will listen
+      /// for trigger messages.
+      /// \return True if the sensor was successfully set to triggered mode.
+      public: bool EnableTriggered(const std::string &_trigger_topic);
+
+      /// \brief Disable triggered mode. The sensor will update as per the set
+      /// update rate.
+      /// \param[in] _update_rate_hz Optional update rate of sensor in Hertz. If
+      /// std::nullopt is passed, the rate parsed from the SDF will be used if
+      /// it is available, otherwise the update rate will be set to zero.
+      public: void DisableTriggered(
+        const std::optional<double> &_update_rate_hz);
```

Comment:
I'm thinking if we can consolidate the `Enable` / `DisableTriggered` functions into one to better match the style of other gz functions for enabling / disabling behavior, something like:

```cpp
// SetTriggred or SetTriggeredEnabled
bool SetTriggered(bool _enable, const std::string &_topic = "") 
```

* when `_enable` is `true`, the `_topic` arg is used
* when `_enable` is `false`, the `_topic` arg does not need to be supplied, and the sensor reverts back to use the update rate value stored in `this->updateRate`. To do this I think we'll need to preserve the value of the `this->updateRate` variable instead of setting it to 0 in triggered mode. If the user wants to a different update rate after disabling triggered mode, they can call `SetUpdateRate` again.

Reply:
Done.